proxy/http/HttpTransact.h

```diff
@@ -571,12 +573,64 @@ class HttpTransact
     ConnectionAttributes *server                 = nullptr;
     ink_time_t now                               = 0;
     ServerState_t state                          = STATE_UNDEFINED;
-    unsigned attempts                            = 0;
-    unsigned simple_retry_attempts               = 0;
-    unsigned unavailable_server_retry_attempts   = 0;
-    ParentRetry_t retry_type                     = PARENT_RETRY_NONE;
+    class Attempts
```

Comment:
Would you make this change separately?

Reply:
I just realized this change is similar to #8776. I guess merging it helps minimizing this PR? @ywkaras

Reply:
This change is already in Yahoo TS.  I think this somehow just got picked up.

Reply:
Sorry for the delay. Yes, I picked this change into HTTP/2 to origin because we had it internally and it fixes a problem we saw in the connect attempts calculation.

Reply:
Just to follow up on this: I just rebased this branch on top of latest 10-Dev. 10-Dev now has Walt's `Attempts` work on it, so that is no longer in this PR.